config/config.go

```diff
@@ -170,3 +173,192 @@ func defaultRunPath() string {
 	}
 	return filepath.Join(homeDir, ".earthly/run")
 }
+
+// UpsertConfig adds or modifies the key to be the specified value.
+// This is saved to disk in your earthly config file.
+func UpsertConfig(config []byte, path, value string) ([]byte, error) {
+	base := &yaml.Node{}
+	yaml.Unmarshal(config, base)
+
+	if base.IsZero() {
+		// Empty file, or a simple comment results in a null document.
+		// Not handled well, so manufacture somewhat acceptable document
+		fullDoc := string(config) + "\n---"
+		yaml.Unmarshal([]byte(fullDoc), base)
+		base.Content = []*yaml.Node{{Kind: yaml.MappingNode}}
+	}
+
+	pathParts := splitPath(path)
+
+	err := validatePath(reflect.TypeOf(Config{}), pathParts)
+	if err != nil {
+		return []byte{}, errors.Wrap(err, "path is not valid")
+	}
+
+	yamlValue, err := valueToYaml(value)
```

Comment:
We parse it as YAML because we needed to support sequences for the `buildkit_additional_args` feature. Turns out this also ends up letting us support complex objects on the other end as inline YAML. I think this is neat.